service/worker/batcher/workflow.go

```diff
@@ -165,7 +165,27 @@ func BatchWorkflow(ctx workflow.Context, batchParams BatchParams) (HeartBeatDeta
 	var result HeartBeatDetails
 	var ac *activities
 	err = workflow.ExecuteActivity(opt, ac.BatchActivity, batchParams).Get(ctx, &result)
-	return result, err
+	if err != nil {
+		return HeartBeatDetails{}, err
+	}
+	return result, attachBatchOperationStats(ctx, result)
+}
+
+const batchWorkflowMemoPrefix = "temporal_sys_batch_workflow_stats_"
+
+var (
+	BatchWorkflowNumSuccessKey = batchWorkflowMemoPrefix + "num_success"
+	BatchWorkflowNumErrorKey   = batchWorkflowMemoPrefix + "num_error"
+)
+
+// attachBatchOperationStats attaches statistics on the number of individual successes and failures to the memo of
+// this workflow.
+func attachBatchOperationStats(ctx workflow.Context, result HeartBeatDetails) error {
+	memo := map[string]interface{}{
+		BatchWorkflowNumSuccessKey: result.SuccessCount,
+		BatchWorkflowNumErrorKey:   result.ErrorCount,
+	}
```

Comment:
I would define a struct, and put it under one key.

Reply:
Done